lib/os/printk.c

```diff
@@ -375,26 +377,32 @@ void vprintk(const char *fmt, va_list ap)
 		}
 	} else {
 		struct out_context ctx = { 0 };
+		k_spinlock_key_t key = k_spin_lock(&lock);
```

Comment:
I support this change, in particular SMP crashes can be incomprehensible, but since the locking is only needed on SMP systems could we macro this so we're not disabling interrupts on UP systems (and affecting latency) while we emit a line of characters?

Reply:
Also, because of latency, we might want to consider making this a Kconfig even on SMP.